packages/vscode-ui5-language-assistant/test/suite/test-utils.ts

```diff
@@ -0,0 +1,56 @@
+import { promises as fs } from "fs";
+import * as vscode from "vscode";
+import * as chai from "chai";
+import { expect } from "chai";
+import deepEqualInAnyOrder from "deep-equal-in-any-order";
+import { TextDocument } from "vscode-languageclient";
+
+chai.use(deepEqualInAnyOrder);
+
+export async function setXMLContent(
+  content: string,
+  xmlPath: string
+): Promise<void> {
+  await fs.writeFile(xmlPath, content);
+  await sleep(1000);
+}
+
+export async function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export function testDiagnostics(
+  docUri: vscode.Uri,
+  expectedDiagnostics: vscode.Diagnostic[]
+): void {
+  const actualDiagnostics = vscode.languages.getDiagnostics(docUri);
+  expect(expectedDiagnostics).to.deep.equalInAnyOrder(actualDiagnostics);
+}
+
+export function getRanges(xmlSnippet: string): vscode.Range[] {
```

Comment:
is this function copy pasted from another package? or parts of?

Reply:
this function has parts from another package